kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -57,6 +59,16 @@ private[kyuubi] class EngineRef private(conf: KyuubiConf, user: String, sessionI
   private val shareLevel: ShareLevel = ShareLevel.withName(conf.get(ENGINE_SHARE_LEVEL))
 
   private val subDomain: Option[String] = conf.get(ENGINE_SHARE_LEVEL_SUB_DOMAIN)
+    .orElse {
+      if (conf.get(ENGINE_POOL_ENABLED)) {
+        // engine.pool.size.threshold is a server-side parameter
+        val poolThreshold = KyuubiConf().loadFileDefaults().get(ENGINE_POOL_SIZE_THRESHOLD)
+        val poolSize = conf.get(ENGINE_POOL_SIZE).min(poolThreshold)
+        Some("engine-pool-" + Random.nextInt(poolSize))
```

Comment:
We can add a TODO here for pooling policies. Currently, it is random and later we can add a sequentially policy such as `AtomicInteger % poolSize`, and son on.

Reply:
make sense.